Helper/Order.php

```diff
@@ -1115,6 +1116,14 @@ public function processNewOrder($quote, $transaction)
                 CreateOrder::E_BOLT_GENERAL_ERROR
             );
         }
+        // When the create_order hook (thread 1) takes a lot of time to execute and returns a timeout, the authorize/payment hook (thread 2) is sent to Magento.
+        // It updates the order prior to the create_order hook (thread 1) process.
+        // This ensures that the returned order in the create_order hook is the latest updated order.
+        $existingOrder =  $this->cartHelper->getOrderById($order->getId());
+        if ($existingOrder->getPayment() && $existingOrder->getPayment()->getMethod() == Payment::METHOD_CODE) {
+            $order = $existingOrder;
+        }
```

Comment:
otherwise some other payment method got in the way and we should throw exception. perhaps. rare case but seems like everything is possible.

Reply:
@danac-gs  I've updated the change as your suggestion. Could you please re-check?